SECURITY.md

```diff
@@ -16,9 +16,40 @@ For example, as of 2024-04-18 the latest release is 0.23.4.
 Therefore 0.23.x, 0.22.x and 0.21.x will be updated, while 0.20.x and 0.19.x
 will not be.
 
-_Note: We use the date of `crates.io` publication when evaluating the security
-policy. For example, while the Rustls 0.20.0 GitHub release note was created
-Jul, 2023 the actual release in `crates.io` was published in Sept. 2021._
+> [!NOTE]
+> We use the date of `crates.io` publication when evaluating the security
+> policy. For example, while the Rustls 0.20.0 GitHub release note was created
+> Jul, 2023 the actual release in `crates.io` was published in Sept. 2021.
+
+### Minimum Supported Rust Version
+
+Our MSRV will be no more recent than 9 versions old, or approximately 12 months.
```

Comment:
I'm a little wary about promising more than about 4-6 versions. Note that Tokio promises 6 months (4 versions) but are usually more conservative in practice. In that sense, I think it often pays off to underpromise and overdeliver.

Also note that (a) tokio-rustls represents about 80% of rustls recent downloads and (b) we bump the tokio-rustls MSRV whenever necessary for a Tokio bump so practically our usability (at least without some amount of pinning) is limited by that.

Reply:
While toward the end we were feeling the pain, we only recently went from an N-19 MSRV (and N-16 before that, in April of this year), and I don't recall that being exceptionally painful. Though there is obviously an opportunity cost there that is difficult to measure.

So I think N-9 is conservative, but not exceptionally so when compared to our recent MSRV history.

My main concern with N-9 right now is that the ecosystem will move more quickly than usual to take advantage of the 2024 edition (which I think we saw with 1.56).